asset_proof.js

```diff
@@ -85,18 +109,119 @@ class AssetProof {
     return true;
   }
 
+  /**
+   * @param {AssetProof} other
+   * @return {boolean}
+   */
+  valueEquals(other) {
+    if (!(other instanceof AssetProof)) {
+      return false;
+    }
+
+    return (
+      other.getId() === this.getId() &&
+      other.getAge() === this.getAge() &&
+      other.getNonce() === this.getNonce() &&
+      other.getInput() === this.getInput() &&
+      other.getHash().length === this.getHash().length &&
+      other.getHash().every((e, i) => e === this.getHash()[i]) &&
+      other.getPrevHash().length === this.getPrevHash().length &&
+      other.getPrevHash().every((e, i) => e === this.getPrevHash()[i])
+    );
+  }
```

Comment:
equivalent to https://github.com/scalar-labs/scalar/blob/master/common/src/main/java/com/scalar/dl/ledger/proof/AssetProof.java#L207